src/components/radio-group/radio-group.tsx

```diff
@@ -75,67 +82,82 @@ export class RadioGroup extends React.Component<Partial<RadioGroupProps>, {}> {
                 data.checked = false;
             });
             this.checkedArray[index].checked = true;
             if (this.props.onChange) {
-                this.props.onChange(value);
+                this.props.onChange(e);
             }
         };
     }
 
-    private createChildrenFromDataSource(): React.ReactNode[] {
-        const childArray: React.ReactNode[] = [];
-        if (Array.isArray(this.props.dataSource)) {
-            this.props.dataSource!.forEach((data, index) => {
-                const props: RadioButtonProps = {
-                    key: index,
-                    value: data.value,
-                    automationId: 'RADIO_BUTTON_' + index,
-                    checked: this.checkedArray[index].checked,
-                    onClick: this.childrenOnClick(index),
-                    disabled: this.props.disabled || data.disabled,
-                    location: this.props.location,
-                    name: this.name
-                };
-                childArray.push(React.createElement(RadioButton, props));
-            });
-        }
-        return childArray;
+    private createChildrenFromDataSource(): Array<React.ReactElement<RadioButton>> {
+        return this.props.dataSource!.map((props, index) => (
+            <RadioButton
+                key={index}
+                value={props.value}
+                data-automation-id={'RADIO_BUTTON_' + index}
+                checked={this.checkedArray[index].checked}
+                onChange={this.childrenOnClick(index)}
+                disabled={this.props.disabled || props.disabled}
+                location={this.props.location}
+                name={this.name}
+                className="radioGroupChild"
+                tabIndex={this.getChildTabIndex(index, this.isGroupChecked)}
+            />
+        ));
     }
 
-    private createChildren(dataArray: any): React.ReactNode[] {
-        const childArray: React.ReactNode[] = [];
-        for (let index = 0; index < dataArray.length; index++) {
-            const data = dataArray[index];
-
-            if (data.type === RadioButton) {
-                const props: RadioButtonProps = {
-                    key: index,
-                    value: data.props.value,
-                    automationId: 'RADIO_BUTTON_' + index,
-                    checked: this.checkedArray[index].checked,
-                    onClick: this.childrenOnClick(index),
-                    disabled: this.props.disabled || data.props.disabled,
-                    location: this.props.location,
-                    name: this.name};
-
-                childArray.push(React.cloneElement(data as React.ReactElement<any>, props));
+    private createChildren(dataArray: React.ReactNode): React.ReactNode[] {
+        return React.Children.map(dataArray, (data, index) => {
+            if (data && typeof data === 'object') {
+                const element = data as React.ReactElement<any>;
```

Comment:
data is of type React.ReactChild (which is of type ReactElement<any> | ReactText).
ReactText is a problem but you are checking that typeof data is object the line before so data is already ReactElement<any> inside this if

Reply:
yet still typescript doesn't realize that i've ruled out ReactText and it needs to be mentioned explicitly